docs/advanced.md

```diff
@@ -0,0 +1,29 @@
+# Advanced
+----------
+
+## Returning Values
+
+Communication with the IPython Kernel is asynchronous, which means that  running Python code blocks any comm messages until the Python code is done. With regards to ITKWidgets this means that getter functions do not "just work" - the Python code cannot complete until the comm message has resolved with the response and the message cannot resolve until the code has completed. This creates a deadlock that prevents the kernel from progressing. This has been a [documented issue](https://github.com/ipython/ipykernel/issues/65) for many years.
```

Comment:
perhaps:

which means that running Python code blocks any comm messages

->

which causes challenges with Python code blocks in Jupyter cells that run synchronously. Multiple comm messages cannot be awaited during the synchronous Python code block execution.